app/models/new_msa_component_event.rb

```diff
@@ -14,12 +15,26 @@ def attributes_to_apply
     {
       name: name,
       component_type: COMPONENT_TYPE::MSA,
+      team_id: team_id,
       environment: environment,
       entity_id: entity_id,
       created_at: created_at
     }
   end
 
+  def team
+    if @team.present?
+      @team
+    elsif team_id.present?
+      @team = Team.find_by_id(team_id)
+    end
+  end
+
+  def team=(team)
+    self.team_id = team.id
+    @team = team
+  end
```

Comment:
If we still need these we could move them to `component.rb` to save defining them twice across both components

Reply:
No wait. Sorry this is an event, not a component boooo